R/create_demographic_lookup.R

```diff
@@ -0,0 +1,432 @@
+      mh ~ "Mental Health",
+      substance ~ "Substance Misuse",
+      medium_cc ~ "Medium Complex Conditions",
+      low_cc ~ "Low Complex Conditions",
+      child_major ~ "Child Major Conditions",
+      adult_major ~ "Adult Major Conditions",
+      comm_living ~ "Assisted Living in the Community",
+      TRUE ~ "Healthy and Low User"
+    )) %>%
+    # Reorder variables
+    dplyr::relocate(.data$demographic_cohort, .after = chi)
+
+  # Write to disk
+  if (write_to_disk == TRUE) {
+    write_rds(demo_lookup,
+      path = glue::glue("{get_slf_dir()}/Cohorts/Demographic_Cohorts_{year}.rds")
+    )
+  }
+
+  return(demo_lookup)
+}
+
+#' Assign Mental Health cohort
+#' @description A record is considered to be in the MH cohort if the recid is 04B. Also,
+#' if the recid is one of 01B, GLS, 50B, 02B or AE2 \strong{and} any of the diagnosis codes start with
+#' F2, F3, F067, F070, F072, F078, or F079
+#'
+#' @param recid A vector of record IDs
+#' @param diag1 A vector of Diagnosis Code 1
+#' @param diag2 A vector of Diagnosis Code 2
+#' @param diag3 A vector of Diagnosis Code 3
+#' @param diag4 A vector of Diagnosis Code 4
+#' @param diag5 A vector of Diagnosis Code 5
+#' @param diag6 A vector of Diagnosis Code 6
+#'
+#' @return A boolean vector indicating whether a given record is in the Mental Health cohort
+#' @family Demographic and Service Use Cohort functions
+assign_mh_cohort <- function(recid, diag1, diag2, diag3, diag4, diag5, diag6) {
+  mh <-
+    # FOR FUTURE: when variable MentalHealthProblemsClientGroup exists and is "Y", mh_cohort = TRUE
+    recid == "04B" |
+      (recid %in% c("01B", "GLS", "50B", "02B", "AE2") &
+        (rowSums(dplyr::across(
+          c("diag1", "diag2", "diag3", "diag4", "diag5", "diag6"),
+          ~ stringr::str_sub(.x, 1, 2) %in%
+            c("F2", "F3")
+        )) > 0 |
+          rowSums(dplyr::across(
+            c("diag1", "diag2", "diag3", "diag4", "diag5", "diag6"),
+            ~ stringr::str_sub(.x, 1, 4) %in%
+              c("F067", "F070", "F072", "F078", "F079")
+          )) > 0))
+  return(mh)
+}
+
+#' Assign Frailty cohort
+#' @description A record is considered to be in the frailty cohort if:
+#' \itemize{
+#'     \item The recid is 01B, 50B, 02B, 04B or AE2 \strong{and}
+#'     \enumerate{\item One of the diagnosis codes starts with W0 or W1
+#'          \item One of the diagnosis codes starts with F00, F01, F02, F03, F05, I61, I63, I64, G20 or G21
+#'          \item One of the diagnosis coes starts with R268 or G22X
+#'    \item The specialty is AB
+#'    \item The significant facility is 1E or 1D
+#'    \item The recid is GLS}}}
+#'
+#' @param recid A vector of record IDs
+#' @param diag1 A vector of Diagnosis Code 1
+#' @param diag2 A vector of Diagnosis Code 2
+#' @param diag3 A vector of Diagnosis Code 3
+#' @param diag4 A vector of Diagnosis Code 4
+#' @param diag5 A vector of Diagnosis Code 5
+#' @param diag6 A vector of Diagnosis Code 6
```

Comment:
See https://cran.r-project.org/web/packages/roxygen2/vignettes/reuse.html specifically
```suggestion
#' @inheritParams assign_mh_cohort 
```
Which would cut down a lot of lines and make it slightly more maintainable.

I think we could probably make much better use of documentation inheritance throughout the package tbh but this seemed a particularly useful case for it.